File: 4_Single_Point_Cloud/single_point_cloud.cc
#include <opencv2/core/core.hpp>
#include <opencv2/imgcodecs.hpp>

#include <opencv2/viz.hpp>
#include <opencv2/calib3d.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/rgbd.hpp>
#include <iostream>
#include <string>

using namespace cv;
using namespace std;

Matx33d getCameraMatrix() {
  // These are the hardcoded camera intrinsics values
  float fx = 532.189488;
  float fy = 532.189488;
  float cx = 318.839986;
  float cy = 244.149197;
  float scale = 10000.0;

  return Matx33d( fx, 0.0, cx,
                  0.0, fy, cy,
                  0.0, 0.0, 1.0);
}

int main() {
  // Create a viz window
  viz::Viz3d viz("Coordinate Frame");

  string color_path = "../../../images/color.png";
  string depth_path = "../../../images/depth.png";

  // Read in the images
  Mat color = imread(color_path, CV_LOAD_IMAGE_COLOR);
  Mat depth = imread(depth_path, CV_LOAD_IMAGE_ANYDEPTH);

  Matx33d camera_matrix = getCameraMatrix();

  Mat depth_cloud;
  rgbd::depthTo3d(depth, camera_matrix, depth_cloud);

  viz.showWidget("cloud", viz::WCloud(depth_cloud, color));
  viz.spin();

  waitKey(0);
  return 0;
}

File: 4_Single_Point_Cloud/single_point_cloud.py
import numpy as np
import cv2

image1_path = '../images/color.png'
image2_path = '../images/depth.png'

window1_title = 'color'
window2_title = 'depth'
window3_title = 'point cloud'

cv2.namedWindow(window1_title, cv2.WINDOW_AUTOSIZE)
cv2.namedWindow(window2_title, cv2.WINDOW_AUTOSIZE)
cv2.namedWindow(window3_title, cv2.WINDOW_AUTOSIZE)

# Read in the images
image1 = cv2.imread(image1_path)
image2 = cv2.imread(image2_path)

# Create the camera calibration matrix
camera_matrix = np.zeros((3,3))
fx = 532.189488 
fy = 532.189488 
cx = 318.839986 
cy = 244.149197
scale = 10000.0

points = []

for v in range(depth_image.height):
  for u in range(depth_image.width):
    Z = depth_image[v,u] / factor
    X = (u - cx) * Z / fx
    Y = (v - cy) * Z / fy
    points.append(
    

points = cv2.depthTo3d()

# --- Display the Images --- #
cv2.imshow(window1_title, image1)
cv2.imshow(window2_title, image2)

cv2.moveWindow(window1_title, 0, 0)
cv2.moveWindow(window2_title, 0, 500)

# Clean up
cv2.waitKey(0)
cv2.destroyAllWindows()
